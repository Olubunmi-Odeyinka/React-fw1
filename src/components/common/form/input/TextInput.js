import React from 'react';
import PropTypes from 'prop-types';
import { Field} from 'formik';

const TextInput = ({name, label, placeholder, touched, error, type, focus, readOnly}) => {
  let wrapperClass = 'form-group';
  if (error && error.length > 0) {
    wrapperClass += " " + 'has-error';
  }

  return (
    <div className={wrapperClass}>
      <label htmlFor={name}>{label}</label>
      <div className="field">
        <Field
          type={type}
          name={name}
          className="form-control"
          placeholder={placeholder}
          readOnly={readOnly}
          autoFocus={focus}/>
        {!readOnly && touched && error && <div className="alert alert-danger">{error}</div>}
      </div>
    </div>
  );
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  touched: PropTypes.bool,
  error: PropTypes.string,
  focus: PropTypes.bool,
  readOnly: PropTypes.bool
};

export default TextInput;
