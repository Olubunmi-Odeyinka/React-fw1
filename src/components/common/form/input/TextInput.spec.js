import React from 'react';
import { shallow } from 'enzyme';
import TextInput from './TextInput';

test('The text input component must render properly', ()=>{
   const props = {
     name: 'm',
     label: 'm',
     touched: false,
     error: '',
     focus: false,
     readOnly: false
  };

   const wrapper = shallow(<TextInput {...props} />);
});
