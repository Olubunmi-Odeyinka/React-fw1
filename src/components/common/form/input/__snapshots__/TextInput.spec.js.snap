// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The text input component input component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    error=""
    focus={true}
    label="Description"
    name="Description"
    readOnly={false}
    touched={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="Description"
>
          Description
</label>,
        <div
          className="field"
>
          <Field
                    autoFocus={true}
                    className="form-control"
                    name="Description"
                    placeholder={undefined}
                    readOnly={false}
                    type={undefined}
          />
</div>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Description",
          "htmlFor": "Description",
        },
        "ref": null,
        "rendered": "Description",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Field
              autoFocus={true}
              className="form-control"
              name="Description"
              placeholder={undefined}
              readOnly={false}
              type={undefined}
/>,
            false,
          ],
          "className": "field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "className": "form-control",
              "name": "Description",
              "placeholder": undefined,
              "readOnly": false,
              "type": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="Description"
>
            Description
</label>,
          <div
            className="field"
>
            <Field
                        autoFocus={true}
                        className="form-control"
                        name="Description"
                        placeholder={undefined}
                        readOnly={false}
                        type={undefined}
            />
</div>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Description",
            "htmlFor": "Description",
          },
          "ref": null,
          "rendered": "Description",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Field
                autoFocus={true}
                className="form-control"
                name="Description"
                placeholder={undefined}
                readOnly={false}
                type={undefined}
/>,
              false,
            ],
            "className": "field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "className": "form-control",
                "name": "Description",
                "placeholder": undefined,
                "readOnly": false,
                "type": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
